package/MDAnalysis/analysis/encore/clustering/ClusteringMethod.py

```diff
@@ -170,7 +170,7 @@ class AffinityPropagation(ClusteringMethod):
 
         def __init__(self,
                      damping=0.9, preference=-1.0,
-                     max_iter=500, convergence_iter=50,
+                     max_iter=500, convergence_iter=50, random_state=0,
```

Comment:
Why is the default random state 0? [The scikit-learn default is "warn".](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.AffinityPropagation.html)

Reply:
If the default state is set as "warn", a warning will be generated and the random_state variable will be set to 0(https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/cluster/_affinity_propagation.py#L147). Hence, it will not make a difference to the best of my understanding.

The primary purpose of setting the default value to 0 is to mimic the function's behaviour before adding the random_state parameter.

Reply:
Perhaps consider the purpose of the warning.

> Hence, it will not make a difference to the best of my understanding.
> 
> The primary purpose of setting the default value to 0 is to mimic the function's behaviour before adding the random_state parameter.

Was the warning emitted before adding the random_state parameter?

Reply:
No, the warning was not emitted before adding the random_state parameter, i.e. Scikit-Learn 0.23. If we go through the warning, it only acknowledges the new parameter `random_state` and specifies its future default value. 

Now to remove the warning(as mentioned in the Issue), we can either pass a constant `random_state` parameter value or add a `random_state` parameter in the class constructor and set the default value explicitly to 0(to mimic the behaviour of previous versions) or None(In this case, calling the function multiple times will produce different results). Please suggest.

Reply:
I think it's best to get the original encore developers' opinions here: cc @mtiberti, @wouterboomsma 

In scikit-learn v0.23 (the latest one) the `random_state` keyword was introduced to control the seed of the random number generator in AffinityPropagation. Up until now the seed was hardcoded as 0. In the yet-to-be released scikit-learn 1.0, the seed defaults to `None`, generating a random initial seed. The current default value of `random_state` in 0.23 is "warn", which warns the user of this upcoming change and then sets the seed to 0. By using a default value of 0, we would be removing this warning.

As sklearn has historically used a seed of 0, Encore's sklearn clustering has historically used 0. With this change in scikit-learn API, we need to now consider whether `encore` should also change.

IMO given that Encore already implements its own C version of AffinityPropagation as default, people who choose to use the sklearn version are doing so because they like sklearn and expect sklearn defaults. However, people may have come to expect certain results from the way we wrap scikit-learn. What do you think, @mtiberti, @wouterboomsma (and to some extent all @MDAnalysis/coredevs ?)

> Now to remove the warning(as mentioned in the Issue)

I see now that this is in the original issue. Correct me if you disagree @jbarnoud, but I don't see this warning as urgently requiring removal; `random_state` is not actually a deprecated feature but just a warning generated due to transitional API by scikit-learn. That is to say, it is not MDAnalysis being backwards that is generating this warning, but scikit-learn deeming it important to warn people about this change.

Reply:
Pinging @MDAnalysis/coredevs @mtiberti @wouterboomsma just because the GSOC student deadline is in 10 days

Reply:
Eagerly waiting for your response!

Reply:
I would go with https://github.com/MDAnalysis/mdanalysis/pull/3178#discussion_r603497072
> the way that I would solve the warnings in tests, and also to make tests pass, is to seed it with 0 in the tests.

For the rest of the code, I'd eventually go with whatever scikit-learn does.

Reply:
@lilyminium I have made the required changes. Please have a look at it.